analyzers/tests/SonarAnalyzer.Test/TestCases/StringLiteralShouldNotBeDuplicated.cs

```diff
@@ -129,3 +129,47 @@ public class SpecialChar
             + "Say \"hello\"";                        // Secondary
     }
 }
+
+namespace SqlNamedParameters
+{
+    public class Program
+    {
+        public void ExecuteSqlCommands()
+        {
+            var userCommand = new SqlCommand("SELECT * FROM Users WHERE Name = @Name");
+            userCommand.AddParameter(new SqlParameter("@Name", "John Doe"));                    // Noncompliant - FP: @Name refers to parameters in different SQL tables.
+            var users = userCommand.ExecuteQuery();                                             // Renaming one does not necessitate renaming of parameters with the same name from other tables.
+
+            var companyCommand = new SqlCommand("SELECT * FROM Companies WHERE Name = @Name");
+            companyCommand.AddParameter(new SqlParameter("@Name", "Contosco"));                 // Secondary - FP
+            var companies = companyCommand.ExecuteQuery();
+
+            var productCommand = new SqlCommand("SELECT * FROM Products WHERE Name = @Name");
+            productCommand.AddParameter(new SqlParameter("@Name", "CleanBot 9000"));            // Secondary - FP
+            var products = productCommand.ExecuteQuery();
+
+            var countryCommand = new SqlCommand("SELECT * FROM Countries WHERE Name = @Name");
+            countryCommand.AddParameter(new SqlParameter("@Name", "Norway"));                   // Secondary - FP
+            var countries = countryCommand.ExecuteQuery();
+        }
+    }
+
+    public class SqlCommand
```

Comment:
For performance reasons, I didn't want to include any SQL libraries or ORMs in the test case, so I created a couple of dummy classes.